backoff/_async.py

```diff
@@ -158,8 +158,10 @@ async def retry(*args, **kwargs):
                                      elapsed >= max_time)
 
                 if giveup_result or max_tries_exceeded or max_time_exceeded:
-                    await _call_handlers(on_giveup, **details)
-                    raise
+                    handler_result = await _call_handlers(on_giveup, **details)
```

Comment:
I don't think this works as is, because `on_giveup` is actually a list of handlers here.  The user facing API accepts either a list or a single handler for `on_giveup`, but in this internal code path it has already been wrapped in a list if it was a single handler function.  The question then is `handler_result` is the return value of _which_ handler.  See the implementation of _call_handlers() to see what I mean.   I liked the thought of having the result value be that of the on_giveup handler, but I doubt think that works in practice.  But let me know if you have any bright ideas.

Reply:
Right, of course! I guess just a bare `return` or `return None` (and not promising to return any values from on_giveup handlers) makes the most sense maybe?

Reply:
Updated to return explicit `None`